testutil/testnode/node_interaction_api.go

```diff
@@ -0,0 +1,143 @@
+package testnode
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/x/payment"
+	"github.com/celestiaorg/celestia-app/x/payment/types"
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+type Context struct {
+	rootCtx context.Context
+	client.Context
+}
+
+func (c *Context) GoContext() context.Context {
+	return c.rootCtx
+}
+
+// LatestHeight returns the latest height of the network or an error if the
+// query fails.
+func (c *Context) LatestHeight() (int64, error) {
+	status, err := c.Client.Status(c.GoContext())
+	if err != nil {
+		return 0, err
+	}
+
+	return status.SyncInfo.LatestBlockHeight, nil
+}
+
+// WaitForHeightWithTimeout is the same as WaitForHeight except the caller can
+// provide a custom timeout.
+func (c *Context) WaitForHeightWithTimeout(h int64, t time.Duration) (int64, error) {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	timeout := time.NewTimer(t)
```

Comment:
[Non Blocking][Suggestion]
Doesn't it make sense to use the context with timeout? to also propagate timeouts if something happens?